### ISSUE TITLE
Type check window instead of document to detect environment

### ISSUE THREAD

**Issue by limdauto** [mochajs/mocha#1764]
I think it's generally accepted that to check whether an environment is node or browser, we should type check `window`, instead of `document`, because sometimes document is mocked for testing purpose.

**Comment by jbnicolai** [comment]
I'd rather use `process.browser`, see https://github.com/mochajs/mocha/blob/master/lib/utils.js#L690

If you'd like to open a PR for this, please do!

**Comment by boneskull** [comment]
`process.browser` please. I tend to stub `window` when unit testing my browserified code on the server.

**Comment by ndhoule** [comment]
Yeah, `process.browser` is the way to go. https://github.com/mochajs/mocha/pull/1727 changed a bunch of these env checks to use `process.browser`, but I pretty much did the bare minimum to get it working so I'm sure there are more.

**Comment by anmuel** [comment]
I'd like to contribute here :)
Was the mentioned TODO the only place to look at? (I doubt it)

**Comment by boneskull** [comment]
@anmuel Thanks.  You could grep around for `window` or `document` and see if there's anything else that needs addressing, but that's about all I know of right now.
